src/PyHyperScattering/SST1RSoXSDB.py

```diff
@@ -1025,72 +1045,46 @@ def loadMd(self, run):
-                    if blval.var() > 0:
-                        warnings.warn(
-                            (
-                                f"While loading {rsoxs} to infill metadata entry for {phs}, found"
-                                f" beginning and end values unequal: {baseline[rsoxs]}.  It is"
-                                " possible something is messed up."
-                            ),
-                            stacklevel=2,
-                        )
+                    md_lookup[key] = [key]
+        ## Find metadata from Tiled and store in PyHyperScattering metadata dictionary
+        for key_name_PHS, key_names_beamline in md_lookup.items():
+            for key_name_beamline in key_names_beamline:
+                if (key_name_PHS in md
+                    and md[key_name_PHS] is not None): 
+                    continue ## If the md is already filled in, no need to try other beamline keys
+                try: md[key_name_PHS] = primary[key_name_beamline].read() ## First try finding metadata in primary stream
                 except (KeyError, HTTPStatusError):
                     try:
-                        md[phs] = primary[md_secondary_lookup[phs]].read()
-                    except (KeyError, HTTPStatusError):
-                        try:
-                            blval = baseline[md_secondary_lookup[phs]]
-                            if (
-                                type(blval) == tiled.client.array.ArrayClient
-                                or type(blval) == tiled.client.array.DaskArrayClient
-                            ):
-                                blval = blval.read()
-                            md[phs] = blval.mean().round(4)
-                            if blval.var() > 0:
-                                warnings.warn(
-                                    (
-                                        f"While loading {md_secondary_lookup[phs]} to infill"
-                                        f" metadata entry for {phs}, found beginning and end"
-                                        f" values unequal: {baseline[rsoxs]}.  It is possible"
-                                        " something is messed up."
-                                    ),
-                                    stacklevel=2,
-                                )
-                        except (KeyError, HTTPStatusError):
+                        baseline_value = baseline[key_name_beamline] ## Next, try finding metadata in baseline
+                        if isinstance(baseline_value, (tiled.client.array.ArrayClient, tiled.client.array.DaskArrayClient)): baseline_value = baseline_value.read() ## For tiled_client.array data types, need to use .read() to get the values
+                        md[key_name_PHS] = baseline_value.mean()
+                        if baseline_value.var() > 0: ## Might need to increase tolerance, so that it does not throw unnecessary warnings for small variations
```

Comment:
```suggestion
                        if baseline_value.var() > 1e-4: # if the value moved significantly during measurement
```
there is an orphan PR that nobody ever reviewed that puts better logic here, this is a placeholder.

Reply:
This PR has since been merged and in the merge commit I added the better logic.